pkg/endpoints/cluster.go

```diff
@@ -57,6 +57,12 @@ func (r Resource) ProxyRequest(request *restful.Request, response *restful.Respo
 	uri := request.PathParameter("subpath") + "?" + parsedURL.RawQuery
 	forwardRequest := r.K8sClient.CoreV1().RESTClient().Verb(request.Request.Method).RequestURI(uri).Body(request.Request.Body)
 	forwardRequest.SetHeader("Content-Type", request.HeaderParameter("Content-Type"))
+	// Copy authorization (for serviceaccount clients) and impersonation headers (for user clients) from the client request to the forwarded request.
+	for k, vs := range request.Request.Header {
```

Comment:
@steveodonovan @a-roberts thoughts on this? Personally I think I'd rather just copy all headers rather than cherry-picking, but looks like we used to do that and removed it for some reason.

Steve, can you remember why you removed this? Was it just the issue with the logs and the Accept header or was there more to it? https://github.com/tektoncd/dashboard/commit/911a81666ee736c6c559c7df26c88b7ea2f26c86#diff-13330ca6ce530de93565ddc93380b446L41

Reply:
Spoke with Steve about this and tested out proxying all headers. The log issue is still present when sending the `Accept` header:
`{"level":"error","msg":"Error: the server was unable to respond with a content type that the client supports"}`

Omitting this seems fine.

We also need to verify the default behaviour without an OIDC authenticator configured on the cluster and ensure any impersonation requests are rejected.